streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamSlidingWindowAggregate.java

```diff
@@ -164,11 +161,13 @@ public void processInOrder(final K key, final V value, final long timestamp) {
 
                     if (endTime < timestamp) {
                         leftWinAgg = next.value;
-                        if (isLeftWindow(next)) {
-                            latestLeftTypeWindow = next.key.window();
-                        }
+                        // update to store the previous record
```

Comment:
This comment doesn't really add anything, it just describes what the code says. Also, don't we need to check that   `windowMaxTimestamp > previousRecordTimestamp` before updating `previousRecordTimestamp` (where `windowMaxTimestamp = next.value.timestamp` -- it would be nice to assign this to a variable with an explicit name to make it clear what `next.value.timestamp` actually means).
Same goes for the below, I guess you could just put the check in a `maybeUpdatePreviousRecordTimestamp()` method and call it from both places

Reply:
I'm not sure we need to check that, I think that by the nature of going through windows forward, the next window we find will always have a max timestamp that's larger than the previous window. Right? Lemme do an example:
Record comes in @30, previous record was at @23, timeDifference = 10. The last window we find with an endTime < timestamp will be 23's left window, where the max record value is 23. Any earlier windows with endTime < timestamp will have a max value less than 23, so we can override them safely